### ISSUE TITLE
Cannot login to an insecure-registry with http:// 

### ISSUE THREAD

**Issue by chengwang86** [vmware/vic#5622]
**VIC version:**

vic 1.1 (actually based on binaries compiled from this branch https://github.com/sflxn/vic/tree/docker-push)

**Deployment details:**
```
vic-machine-linux create --force --name=vch-1 --insecure-registry http://kang.eng.vmware.com --target=root:pwd@192.168.213.129 --debug 1 --image-store=datastore1 --no-tlsverify --no-tls
```

**Steps to reproduce:**
after deploying the vch, run `sudo docker -H ... login http://kang.eng.vmware.com` or `sudo docker -H ... kang.eng.vmware.com`.  Here `kang.vmware.com` is a harbor instance deployed on a Ubuntu desktop VM.

**Actual behavior:**
Both return the same error msg: `rror response from daemon: Head https://kang.eng.vmware.com/v2/: dial tcp 10.17.109.201:443: getsockopt: connection refused`. However, both cmds suceeed on vanilla docker. 

**Expected behavior:**
`sudo docker login kang.eng.vmware.com` succeeds.

**Logs:**
[container-logs.zip](https://github.com/vmware/vic/files/1112605/container-logs.zip)

**Additional details as necessary:**

**Comment by chengwang86** [comment]
From persona log
```
Jun 29 2017 15:15:20.557Z DEBUG Calling POST /v1.25/auth
Jun 29 2017 15:15:20.557Z DEBUG form data: {"password":"*****","serveraddress":"kang.eng.vmware.com","username":"cheng"}
```
It seems like the form data only have `kang.eng.vmware.com`, no matter I run `docker login kang....` or `docker login http://kang...` 

However, from vch init.log,
```
guestinfo.vice./registry/insecure_registries~, value: \"http://kang.eng.vmware.com\"
```
so there is a mismatch between the url value we store as the insecure-registry and the actual form data used for login authentication.

**Comment by mhagen-vmware** [comment]
Pull is not working anymore on insecure registries as well:
```
docker -H 192.168.254.155:2375 pull 192.168.254.1:5000/busybox
Using default tag: latest
Error response from daemon: Head https://192.168.254.1:5000/v2/: http: server gave HTTP response to HTTPS client
```
on VIC 1.1.1:
```
docker -H 192.168.254.148:2375 pull 192.168.254.1:5000/busybox
Using default tag: latest
Pulling from busybox
27144aa8f1b9: Pull complete 
a3ed95caeb02: Pull complete 
Digest: sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642
Status: Downloaded newer image for busybox:latest
```

**Comment by gigawhitlocks** [comment]
Notice how the path is changed from http:// to https:// between the user input and the error message. There's something wonky about the way that we sanitize these paths and compare them in the actual list. Namely, it seems to be done ad-hoc everywhere a path is input (so far as I can tell investigating the Whitelist test also) and then the comparison is a plain string comparison, so if the URL is not standardized/sanitized in exactly the right way, we make an assumption in one place or another about what it "should" be. It looks like here at some point the `http` gets stripped and `https` is assumed and prepended at a later time.

**Comment by gigawhitlocks** [comment]
Currently experiencing similar behavior surrounding https://same_ip/ vs https://same_ip:443, which should be the same, not being treated as such

**Comment by gigawhitlocks** [comment]
e.g., here is one location with a `TODO` from @jzt indicating the same as my comment above https://github.com/vmware/vic/blob/master/lib/apiservers/engine/backends/system.go#L322

**Comment by gigawhitlocks** [comment]
I'm going to assign this to myself under the assumption that fixing the nightly `21-1-Whitelist` (#5538) as well as this issue requires better standardization of the registry URL at whitelist insertion time in `vic-machine`.

**Comment by gigawhitlocks** [comment]
Closed by accident

**Comment by gigawhitlocks** [comment]
This may be resolved via #5702, needs verification

**Comment by gigawhitlocks** [comment]
Verified via #5756
